geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/guard/GraduatedQueryGuard.scala

```diff
@@ -8,76 +8,61 @@
+    disabled = GraduatedQueryGuard.disabled(sft.getTypeName)
+    if (disabled) {
+      logger.info(s"This guard is disabled for schema '${sft.getTypeName}' via system property")
     }
+    // let any errors bubble up and disable this guard
+    guardLimits = buildLimits(ConfigFactory.load().getConfigList(s"$ConfigPath.${sft.getTypeName}"))
   }
 
-  private def guardInner(strategy: QueryStrategy): Option[IllegalArgumentException] = {
-    val intervalsOption: Option[FilterValues[Bounds[ZonedDateTime]]] = strategy.values.collect { case v: TemporalIndexValues => v.intervals }
-    val spatialBoundsOption: Option[Seq[(Double, Double, Double, Double)]] = strategy.values.collect { case v: SpatialIndexValues => v.spatialBounds }
-
-    val spatialExtent = spatialBoundsOption match {
-      case None => 360.0 * 180.0  // Whole world
-      case Some(seq) => seq.map { case  (lx: Double, ly: Double, ux: Double, uy: Double) =>
-        (ux-lx) * (uy-ly)
-      }.sum
-    }
+  override def rewrite(query: Query): Unit = {}
 
-    intervalsOption match {
-      // The None case reflects a query without a temporal extent or a full-table scan on a table with a temporal extent
-      // The full-table scan block should be used to avoid the later case.
-      // Here a none is returned indicating that this guard does not stop the query
-      case None => None
-      case Some(intervals) =>
-        if (!intervals.forall(_.isBoundedBothSides) || intervals.isEmpty) {
-          Some(new IllegalArgumentException("At least one part of the query is temporally unbounded.  Add bounded temporal constraints."))
-        } else {
-          val queryDuration = duration(intervals.values)
-          guardLimits.find( _.sizeLimit >= spatialExtent).flatMap {
-            limit =>
-              if (queryDuration > limit.durationLimit) {
-                Some(new IllegalArgumentException(
-                  s"Query of this size exceeds maximum allowed filter duration of ${limit.durationLimit}: " +
-                    s"${filterToString(strategy.filter.filter)}. " +
-                    s"Try a small spatial extent or a shorter time range."))
-              } else {
-                None
-              }
+  override def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = {
+    val msg = if (disabled || !strategy.index.isInstanceOf[SpatioTemporalIndex[_, _]]) { None } else {
+      val values = strategy.values.collect {
+        case v: SpatialIndexValues with TemporalIndexValues => (v.spatialBounds, v.intervals)
+      }
+      values match {
+        case None => Some("Query does not have a temporal filter")
+        case Some((s, i)) =>
+          val spatialExtent = s.map { case (lx, ly, ux, uy) => (ux - lx) * (uy - ly) }.sum
+          val limit = guardLimits.find(_.sizeLimit >= spatialExtent).getOrElse {
+            throw new IllegalStateException(
+              s"Invalid extents/limits: ${s.mkString(", ")} / ${guardLimits.mkString(", ")}")
```

Comment:
Shouldn't the guardLimits have an unbounded limit?  Does this happen?

Reply:
yeah i don't think this should ever happen, so i used the IllegalStateException to indicate something when quite wrong. seemed better than just a raw `.get`. i could maybe envision some really complex series of polygon filters that, in aggregate, exceeded Int.MaxValue degrees and flipped negative?

Reply:
+1; just making sure I understood.

Seems like we could add the "//  CANNOT GET HERE" comment:).